tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/statements/matchstmt/varbindingpatternmatchpattern/ErrorBindingPatternTest.java

```diff
@@ -42,6 +50,16 @@ public void setup() {
                 "/error_binding_pattern_negative.bal");
     }
 
+    @Test
+    public void testNegativeSemantics() {
+        CompileResult buildError = BCompileUtil.compile("test-src/statements/matchstmt/" +
+                "varbindingpatternmatchpattern/error_type_ref_negative.bal");
+        int i = -1;
+        BAssertUtil.validateError(buildError, ++i, "invalid error binding pattern with type 'other'", 18, 11);
```

Comment:
If it is a straightforward fix, shall we stop logging this error with "other"? Errors with "other" are usually because of other errors and don't add value to the user?

Reply:
done